embedded_rtic/examples/sonars.rs

```diff
@@ -0,0 +1,440 @@
+        Forward,
+        Left,
+        Right,
+    }
+
+    #[task(binds = GPIOTE,priority = 4, local=[echo,sender,previous_time,echo2,sender2,previous_time2,echo3,sender3,previous_time3,times],shared = [gpiote])]
+    /// Reads the echo pin
+    ///
+    /// When ever the echo pin goes high or low this interrupt is triggered.
+    /// On the falling edge this interrupt computes the "width" in time
+    /// of the square wave thus allowing us to compute the time it took
+    /// to echo back to us.
+    fn echo(mut cx: echo::Context) {
+        // Get time ASAP for highest granularity
+        let time = Mono::now();
+        let zero: Instant = Instant::from_ticks(0);
+
+        // Array to store triggered sonars
+        let mut triggered_sonars = [None; NUM_SONARS];
+
+        // Check which sonar triggered the event and store it in the array
+        let debug_time = time.duration_since_epoch().to_micros();
+        cx.shared.gpiote.lock(|gpiote| {
+            if gpiote.channel0().is_event_triggered() {
+                info!("Interrupt on channel 0, {:?}", debug_time);
+                triggered_sonars[0] = Some(SonarChannel::Forward);
+            }
+            if gpiote.channel1().is_event_triggered() {
+                info!("Interrupt on channel 1, {:?}", debug_time);
+                triggered_sonars[1] = Some(SonarChannel::Left);
+            }
+            if gpiote.channel2().is_event_triggered() {
+                info!("Interrupt on channel 2, {:?}",debug_time);
+                triggered_sonars[2] = Some(SonarChannel::Right);
+            }
+        });
+
+        // Print messages for each triggered sonar
+        for sonar in triggered_sonars.iter().filter_map(|x| *x) {
+            match sonar {
+                SonarChannel::Forward => {
+                    //info!("Event triggered by Sonar 1 (Forward)");
+                    if cx.local.times[0] == zero {
+                        cx.local.times[0] = time;
+                    } else {
+                        let distance = time.checked_duration_since(cx.local.times[0]).unwrap().to_micros() / 56;
+                        match cx.local.sender.try_send(distance as u32) {
+                            Ok(_) => {}
+                            _ => {
+                                error!("Distance FORWARD did not fit in to the buffer, scheduling is probably broken.");
+                                // This is not fatal so we simply continue with our life
+                            }
+                        }
+                        cx.local.times[0] = zero;
+                    }
+                }
+                SonarChannel::Left => {
+                    //info!("Event triggered by Sonar 2 (Left)");
+                    if cx.local.times[1] == zero {
+                        cx.local.times[1] = time;
+                    } else {
+                        let distance2 = time.checked_duration_since(cx.local.times[1]).unwrap().to_micros() / 56;
```

Comment:
Distance 2 and 3 are not needed here as they are defined in the block you can just re-use the same variable name so i think 
```rust
let distance = ....
```
for all of these would be more readable